modal/cli/secret.py

```diff
@@ -48,32 +50,69 @@ async def list_(env: Optional[str] = ENV_OPTION, json: bool = False):
 @secret_cli.command("create", help="Create a new secret.")
 @synchronizer.create_blocking
 async def create(
-    secret_name,
-    keyvalues: list[str] = typer.Argument(..., help="Space-separated KEY=VALUE items"),
+    secret_name: str,
+    keyvalues: Optional[list[str]] = typer.Argument(default=None, help="Space-separated KEY=VALUE items."),
     env: Optional[str] = ENV_OPTION,
+    from_dotenv: Optional[Path] = typer.Option(default=None, help="Path to a .env file to load secrets from."),
+    from_json: Optional[Path] = typer.Option(default=None, help="Path to a JSON file to load secrets from."),
     force: bool = typer.Option(False, "--force", help="Overwrite the secret if it already exists."),
 ):
     env = ensure_env(env)
     env_dict = {}
-    for arg in keyvalues:
+
+    for arg in keyvalues or []:
         if "=" in arg:
             key, value = arg.split("=", 1)
             if value == "-":
                 value = get_text_from_editor(key)
             env_dict[key] = value
         else:
             raise click.UsageError(
-                """Each item should be of the form <KEY>=VALUE. To enter secrets using your $EDITOR, use `<KEY>=-`.
+                """Each item should be of the form <KEY>=VALUE. To enter secrets using your $EDITOR, use `<KEY>=-`. To
+enter secrets from environment variables, use `<KEY>="$ENV_VAR"`.
 
 E.g.
 
 modal secret create my-credentials username=john password=-
+modal secret create my-credentials username=john password="$PASSWORD"
 """
             )
 
+    if from_dotenv:
+        if not from_dotenv.is_file():
+            raise click.UsageError(f"Could not read .env file at {from_dotenv}")
+
+        try:
+            from dotenv import dotenv_values
+        except ImportError:
+            raise ImportError(
+                "Need the `python-dotenv` package installed. You can install it by running `pip install python-dotenv`."
+            )
+
+        try:
+            env_dict.update(dotenv_values(from_dotenv))
+        except Exception as e:
+            raise click.UsageError(f"Could not parse .env file at {from_dotenv}: {e}")
+
+    if from_json:
+        if not from_json.is_file():
+            raise click.UsageError(f"Could not read JSON file at {from_json}")
+
+        try:
+            with from_json.open("r") as f:
+                env_dict.update(json.load(f))
+        except Exception as e:
+            raise click.UsageError(f"Could not parse JSON file at {from_json}: {e}")
+
     if not env_dict:
         raise click.UsageError("You need to specify at least one key for your secret")
 
+    for k, v in env_dict.items():
+        if not isinstance(k, str) or not k:
+            raise click.UsageError(f"Invalid key for secret '{k}'")
```

Comment:
Nit, but I think I'd otherwise misread this as reporting the secret's _name_.

```suggestion
            raise click.UsageError(f"Invalid key: '{k}'")
```